CMakeLists.txt

```diff
@@ -15,6 +15,9 @@ set(PYTHON_SUPPORTED_VERSIONS "3.8" "3.9" "3.10" "3.11")
 # Supported NVIDIA architectures.
 set(CUDA_SUPPORTED_ARCHS "7.0;7.5;8.0;8.6;8.9;9.0")
 
+# used when building pytorch-related extensions
+set(TORCH_CUDA_ARCH_LIST "7.0;7.5;8.0;8.6;8.9;9.0")
```

Comment:
note that pytorch 2.2.0 has 9.0a support by default:

https://github.com/pytorch/pytorch/blob/19d27a13ea052230d9fb565a5b82e683e28d1697/Dockerfile#L60

while our docker image does not support 9.0a

Reply:
Doesn't this make our build system always compile the CUDA kernels for all architectures?

If I remember correctly, we only compiled the kernels for a single architecture by detecting the equipped GPUs on the user machine (I'm not sure this is still true after we changed our build system to CMake though), to reduce the compile time. Exceptionally, we targeted all architectures when building docker images or pypi wheels.

Reply:
This is used in docker image. It seems this CMake inherit the build architecture from pytorch by default, and so I have to change it (to avoid 9.0a architecture that is not supported in docker nvcc).